File: backend/src/AnimalFarm.Domain/Entities/Animal.cs
using AnimalFarm.Domain.Enums;

namespace AnimalFarm.Domain.Entities;

public class Animal
{
    public int Id { get; set; }
    public string InventoryNumber { get; set; } = string.Empty;
    public Gender Gender { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime ArrivalDate { get; set; }
    public int ArrivalAgeMonths { get; set; }
    public int BreedId { get; set; }
    public Breed Breed { get; set; } = null!;
    public int? ParentAnimalId { get; set; }
    public Animal? ParentAnimal { get; set; }
    public ICollection<Animal> ChildAnimals { get; set; } = new List<Animal>();
    public ICollection<Weighting> Weightings { get; set; } = new List<Weighting>();
}

File: frontend/src/types/animal.ts
export interface AnimalType {
  id: number;
  name: string;
}

export interface CreateAnimalTypeRequest {
  name: string;
}

export interface Breed {
  id: number;
  name: string;
  animalTypeId: number;
  animalTypeName: string;
}

export interface CreateBreedRequest {
  name: string;
  animalTypeId: number;
}

export interface Animal {
  id: number;
  inventoryNumber: string;
  gender: string;
  name: string;
  arrivalDate: string;
  arrivalAgeMonths: number;
  breedId: number;
  breedName: string;
  parentAnimalId: number | null;
  parentAnimalName: string | null;
}

export interface CreateAnimalRequest {
  inventoryNumber: string;
  gender: string;
  name: string;
  arrivalDate: string;
  arrivalAgeMonths: number;
  breedId: number;
  parentAnimalId: number | null;
}

export interface Weighting {
  id: number;
  animalId: number;
  animalName: string;
  userId: string;
  userName: string;
  date: string;
  weightKg: number;
}

export interface CreateWeightingRequest {
  animalId: number;
  date: string;
  weightKg: number;
}

export interface UserDto {
  id: string;
  userName: string;
  email: string;
  isActive: boolean;
  roles: string[];
}
